### DISCUSSION TITLE
difference between last_picker and command_palette

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#9421] [General]
On doing `space."'"` and `space."?"` helix opens `last_picker` and `command_palette` correspondingly. But both of them looks alike. Is there something different about them??

**Comment by the-mikedavis** [comment]
`last_picker` opens whichever picker you opened last. For example if you open the file picker (`<space>f`), close it and then use `<space>'`, the file picker will be re-opened.

**Reply by ghost** [reply]
its kind of weird feature, isn't it?

**Reply by gabydd** [reply]
It's actually extremely useful because we remember the state that the picker was in letting you do something like a global search  filter the files look choose one of the results look around decide you need a different result and then press ``<space>'`` and be right to where you need to be

**Reply by gabydd** [reply]
Also works after you do a goto references ``gf`` which can be very useful when refactoring across multiple files
